src/fips_check.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * This program is released under the Common Public License V1.0
+ *
+ * You should have received a copy of Common Public License V1.0 along with
+ * with this program.
+ *
+ * Copyright IBM Corp. 2020
+ */
+
+
+#ifdef ICA_FIPS
+
+#include <errno.h>
+#include <openssl/crypto.h>
+#include <openssl/hmac.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <dlfcn.h>
+
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+#define HMAC_PREFIX "."
+#ifndef HMAC_SUFFIX
+#define HMAC_SUFFIX ".hmac"
+#endif
+#define READ_BUFFER_LENGTH 16384
+
+static const char hmackey[] = "00000000000000000000000000000000";
```

Comment:
do we really want to encode the key in ascii ? ascii 0 == 48. I think we should use static cons tunsigned char hmacckey[32] = {0} i.e. the all-zero-key, instead of an all-48-key.

Reply:
As our distributor will encode the known HMAC in the .hmac file in the same way as string, I would prefer to leave it as a string here as well. But you are right about the init value, which is misleading. What about just
> static const char hmackey[32];

without an init value. However, it's more obvious for Distributors how to insert the known HMAC key when we leave it as it is. To really provide this as a placeholder, we could even do it like:
>  static const char hmackey[] = "000102030405060708090A0B0C0D0E0F";

And yes, this would be only 16 bytes ... maybe we should then show an example with 32 bytes.

Reply:
Okay, but if you encode the hmac key as a string (like the known mac in the file), you should process it with OPENSSL_hexstr2buf before using it (as you do for the known mac). In that case it would be okay to initialize hmackey to a string of "0...". Id use a 32 byte all zero key as default value.

Reply:
What i mean is: afaict right now, the values of the key's bytes are the values from the ascii table for the corresponding letter in the string but i think it should be the hex value encoded by two letters of the string.

Reply:
OK, now using OPENSSL_hexstr2buf. I also noticed that the recommended key size for HMAC-SHA256 is even 64 bytes. So I'm now providing a "zero"-key with 64 bytes:
> static const char hmackey[] =
> 	"0000000000000000000000000000000000000000000000000000000000000000"
> 	"0000000000000000000000000000000000000000000000000000000000000000";

Distributors are however free to change this.